pkg/utils/map_test.go

```diff
@@ -55,6 +55,13 @@ func TestMergeMapNested(t *testing.T) {
 			nil,
 			assert.Error,
 		},
+		{
+			"simple merge overlapping no overwrite nil value",
```

Comment:
Added a few unit tests around nil values to make sure the changes in map.go are OK.